src/ServerMainComponent.cpp

```diff
@@ -545,8 +545,16 @@ void ServerMainComponent::timerCallback()
 					if (nextWorkingSet->get_control_function()->get_address() != activeWorkingSetMasterAddress)
 					{
 						activeWorkingSetMasterAddress = nextWorkingSet->get_control_function()->get_address();
-						activeWorkingSetDataMaskObjectID = std::static_pointer_cast<isobus::WorkingSet>(nextWorkingSet->get_working_set_object())->get_active_mask();
-						newWorkingSetFound = true;
+						if (nextWorkingSet->get_working_set_object())
+						{
+							activeWorkingSetDataMaskObjectID = std::static_pointer_cast<isobus::WorkingSet>(nextWorkingSet->get_working_set_object())->get_active_mask();
```

Comment:
Calling `get_working_set_object` twice means it will have to search the object tree 2x instead of once, might want to just save the result in a variable rather than search for it twice

Reply:
Good catch, fixed it, thanks for spotting!